zipper/protocols/irondb/irondb_helpers.go

```diff
@@ -0,0 +1,125 @@
+package irondb
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+)
+
+// graphiteExprListToIronDBTagQuery - converts list of Graphite Tag expressions to IronDB Tag query
+// e.g. ["name=~cpu\..*", "tag1!=value1"] => and(__name:/cpu\..*/,not(tag1:value1))
+func graphiteExprListToIronDBTagQuery(exprList []string) string {
+	//r := make([]string, len(exprList))
+	var r []string
+
+	for _, expr := range exprList {
+		// remove escaped quotes
+		expr = strings.ReplaceAll(expr, "\"", "")
+		eq_idx := strings.Index(expr, "=")
+		if eq_idx > 0 {
+			tag_name := expr[:eq_idx]
+			ne_idx := strings.Index(expr, "!")
+			if ne_idx != -1 {
+				tag_name = expr[:ne_idx]
+			}
+			if tag_name == "name" {
+				tag_name = "__name"
+			}
+			if ne_idx == -1 {
```

Comment:
It seems that the only thing that is dfferent is the operation type (and -> not), otherise then and else are the same.

Reply:
reworked that, should be better now